File: core/baker-interface/src/main/scala/com/ing/baker/runtime/scaladsl/EventResolutions.scala
package com.ing.baker.runtime.scaladsl

import com.ing.baker.runtime.common.SensoryEventStatus

import scala.concurrent.Future

case class EventResolutions(
                             resolveWhenReceived: Future[SensoryEventStatus],
                             resolveWhenCompleted: Future[SensoryEventResult]
) extends EventResolutionsF[Future](resolveWhenReceived, resolveWhenCompleted)

File: core/baker-interface-in-memory/src/main/kotlin/com/ing/baker/runtime/inmemory/EventResolutions.kt
package com.ing.baker.runtime.inmemory

import com.ing.baker.runtime.common.SensoryEventStatus
import com.ing.baker.runtime.javadsl.SensoryEventResult
import kotlinx.coroutines.Deferred

/**
 * Deferred resolutions for the deprecated callback-style fireEvent API.
 */
data class KotlinEventResolutions(
    val resolveWhenReceived: Deferred<SensoryEventStatus>,
    val resolveWhenCompleted: Deferred<SensoryEventResult>
)

